src/herder/TransactionQueue.h

```diff
@@ -230,15 +234,37 @@ class TransactionQueue
 
     bool isFiltered(TransactionFrameBasePtr tx) const;
 
-    std::unique_ptr<TxQueueLimiter> mTxQueueLimiter;
+    // Snapshots to use for transaction validation
+    ImmutableValidationSnapshotPtr mValidationSnapshot;
+    SearchableSnapshotConstPtr mBucketSnapshot;
+
+    TxQueueLimiter mTxQueueLimiter;
     UnorderedMap<AssetPair, uint32_t, AssetPairHash> mArbitrageFloodDamping;
 
     UnorderedMap<Hash, TransactionFrameBasePtr> mKnownTxHashes;
 
     size_t mBroadcastSeed;
 
+    mutable std::recursive_mutex mTxQueueMutex;
```

Comment:
is recursive_mutex required here? perhaps mutex is more appropriate?

Reply:
I used a `recursive_mutex` because the timer asynchronously calls the private function `broadcast`, so `broadcast` needs to grab the lock. That causes a problem because other public functions also call `broadcast`. That being said, I think I can refactor these functions to fix this. The async call really shouldn't invoke a private function, as it makes control flow harder to reason about.

Reply:
I see. If it's not too painful to refactor, I think it might be cleaner. I suggested a simple mutex, because it gives you more control and visibility of the code flow. For example, passing a lock guard reference as an argument is less error-prone than grabbing a recursive mutex (something we also learned with background overlay).

Reply:
Done in 7bb9da0cac81c5793add740e9dfd5a5eeb342871